### ISSUE TITLE
[AMDGPU] CGBuiltin - bad handling of builtin_amdgcn_fence

### ISSUE THREAD

**Issue by RKSimon** [llvm/llvm-project#56832]
As mentioned here: https://reviews.llvm.org/D80804#inline-1081007

The `LLVM_FALLTHROUGH` handling of `builtin_amdgcn_fence` is missing:
```cpp
  case AMDGPU::BI__builtin_amdgcn_fence: {
    if (ProcessOrderScopeAMDGCN(EmitScalarExpr(E->getArg(0)),
                                EmitScalarExpr(E->getArg(1)), AO, SSID))
      return Builder.CreateFence(AO, SSID);
    LLVM_FALLTHROUGH;
  }
  case AMDGPU::BI__builtin_amdgcn_atomic_inc32:
  case AMDGPU::BI__builtin_amdgcn_atomic_inc64:
  case AMDGPU::BI__builtin_amdgcn_atomic_dec32:
  case AMDGPU::BI__builtin_amdgcn_atomic_dec64: {
    unsigned BuiltinAtomicOp;
    llvm::Type *ResultType = ConvertType(E->getType());

    switch (BuiltinID) {
    case AMDGPU::BI__builtin_amdgcn_atomic_inc32:
    case AMDGPU::BI__builtin_amdgcn_atomic_inc64:
      BuiltinAtomicOp = Intrinsic::amdgcn_atomic_inc;
      break;
    case AMDGPU::BI__builtin_amdgcn_atomic_dec32:
    case AMDGPU::BI__builtin_amdgcn_atomic_dec64:
      BuiltinAtomicOp = Intrinsic::amdgcn_atomic_dec;
      break;
    }
```

**Comment by arsenm** [comment]
I think the fallthrough handling and check here is dead code. The order and scope should always be constants and should be enforced by the builtin definition

**Comment by arsenm** [comment]
https://reviews.llvm.org/D131183

**Comment by arsenm** [comment]
Fixed by c5b36ab1d6a667554bf369c34e51d02add039d16
